packages/application-components/src/components/modal-pages/form-modal-page/form-modal-page.js

```diff
@@ -0,0 +1,90 @@
+import ModalPageTopBar from '../internals/modal-page-top-bar';
+import ModalPageHeader from '../internals/modal-page-header';
+import { ContentWrapper } from '../internals/modal-page.styles';
+import messages from '../internals/messages';
+
+const useLabel = (label, intl) =>
+  typeof label === 'string' ? label : intl.formatMessage(label);
+
+const FormModalPage = props => (
+  <ModalPage
+    level={props.level}
+    title={props.title}
+    isOpen={props.isOpen}
+    zIndex={props.zIndex}
+    onClose={props.onClose}
+    baseZIndex={props.baseZIndex}
+    parentSelector={props.parentSelector}
+  >
+    <ModalPageTopBar
+      onClose={props.onClose}
+      currentPathLabel={props.topBarCurrentPathLabel || props.title}
+      previousPathLabel={props.topBarPreviousPathLabel}
+    />
+    <ModalPageHeader
+      title={props.title}
+      subtitle={props.subtitle}
+      customControls={props.customControls}
+      labelPrimaryButton={useLabel(props.labelPrimaryButton, props.intl)}
+      labelSecondaryButton={useLabel(props.labelSecondaryButton, props.intl)}
+      onPrimaryButtonClick={props.onPrimaryButtonClick}
+      onSecondaryButtonClick={props.onSecondaryButtonClick}
+      isPrimaryButtonDisabled={props.isPrimaryButtonDisabled}
+      dataAttributesPrimaryButton={props.dataAttributesPrimaryButton}
+      dataAttributesSecondaryButton={props.dataAttributesSecondaryButton}
+    />
+    <ContentWrapper>{props.children}</ContentWrapper>
+  </ModalPage>
+);
+FormModalPage.displayName = 'FormModalPage';
+FormModalPage.propTypes = {
+  level: PropTypes.number,
+  title: PropTypes.string.isRequired,
+  zIndex: PropTypes.number,
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func,
+  children: PropTypes.node.isRequired,
+  baseZIndex: PropTypes.number,
+  parentSelector: PropTypes.string,
+  // TopBar props
+  topBarCurrentPathLabel: PropTypes.string,
+  topBarPreviousPathLabel: PropTypes.string,
+  // Header props
+  subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  customControls: PropTypes.node,
+  labelSecondaryButton: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      defaultMessage: PropTypes.string.isRequired,
+    }),
+  ]).isRequired,
+  labelPrimaryButton: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      defaultMessage: PropTypes.string.isRequired,
+    }),
+  ]).isRequired,
+  onPrimaryButtonClick: PropTypes.func.isRequired,
+  onSecondaryButtonClick: PropTypes.func,
+  isPrimaryButtonDisabled: PropTypes.bool,
+  dataAttributesPrimaryButton: PropTypes.object,
+  dataAttributesSecondaryButton: PropTypes.object,
+  // injected
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }).isRequired,
+};
+FormModalPage.defaultProps = {
+  labelPrimaryButton: messages.confirm,
+  labelSecondaryButton: messages.cancel,
+};
+
+FormModalPage.Intl = messages;
```

Comment:
Careful, you're exposing also messages that should not be used for the button labels, e.g. "back".